packages/frontend/app/(dashboard)/course/[cid]/components/chatbot/Chatbot.tsx

```diff
@@ -210,16 +210,18 @@ const Chatbot: React.FC<ChatbotProps> = ({
     const result: ChatbotAskResponse = await query()
 
     const answer = result ? result.answer : "Sorry, I couldn't find the answer"
+    const { thinkText, cleanAnswer } = parseThinkBlock(answer)
     const sourceDocuments = result ? result.sourceDocuments : []
     setMessages((prevMessages: Message[]) => [
       ...prevMessages,
       { type: 'userMessage', message: input },
       {
         type: 'apiMessage',
-        message: answer,
+        message: thinkText ? cleanAnswer : answer,
```

Comment:
if there's no match, cleanAnswer contains the original answer no matter what, correct?

can't we just
```suggestion
        message: cleanAnswer
```

Reply:
yeah that's true, but a part of me wants to leave it as is for some small reasons:
- I think it's a little more clear about what's happening
- In case `parseThinkBlock` gets changed in the future so that cleanAnswer doesn't just default to the old answer, this implementation would be a bit less bug-prone i guess